projects/plugins/jetpack/extensions/blocks/subscriptions/view.js

```diff
@@ -14,33 +14,33 @@ domReady( function () {
 	if ( ! form.payments_attached ) {
 		form.payments_attached = true;
 		form.addEventListener( 'submit', function ( event ) {
+			event.preventDefault();
+
 			const email = form.querySelector( 'input[type=email]' ).value;
-			const email_clause = email ? `&email=${ encodeURIComponent( email ) }` : '';
-			const post_id = form.querySelector( 'input[name=post_id]' )?.value;
-			const post_id_clause = post_id ? `&post_id=${ post_id }` : '';
-			const tier_id = form.querySelector( 'input[name=tier_id]' )?.value;
-			const tier_id_clause = tier_id ? `&tier_id=${ tier_id }` : '';
 
 			if ( form.resubmitted || ! email ) {
 				return;
 			}
-			event.preventDefault();
```

Comment:
I am not sure about what is the `form.resubmitted` use-case (and I could not find any data on it), but 
```
event.preventDefault();
```
should maybe stay after that check? So it can be handled natively?

Reply:
Sure! I'll debug a bit and put it down if it affects.

Reply:
I didn't really find a way to test `resubmitted` quickly (would be curious to learn tho) so I just put it back where it was, not a biggie.